numba/cuda/cudaimpl.py

```diff
@@ -992,6 +992,7 @@ def _generic_array(context, builder, shape, dtype, symbol_name, addrspace,
     other_supported_type = (
         isinstance(dtype, (types.Record, types.Boolean))
         or isinstance(data_model, models.StructModel)
+        or dtype == types.float16
```

Comment:
What do you think about adding `types.float16` to `types.number_domain` instead of making it a special case here?